### PR TITLE
Update all Dependencies

### PR DESCRIPTION
**Pull request by DRSchlaubi**
- Fix bug with gateway connection related to Ktor 1.5
- Fix bug with kotlinx.atomicfu 0.15

### CODE REVIEW

File: gateway/src/main/kotlin/handler/HeartbeatHandler.kt [hunk 1]
```diff
@@ -54,4 +54,14 @@ internal class HeartbeatHandler(
             ticker.stop()
         }
     }
+
+    // These two methods avoid a bug with the kotlinx.atomicfu compiler plugin
```
Comment (BartArys) [inline gateway/src/main/kotlin/handler/HeartbeatHandler.kt:58]: 'a bug' is not descriptive enough, if we have any issue on the matter reference to it, or reference to this PR.

Consider putting these comments on each function individually as well. Code has a history of drifting apart and being reorganized.

Comment (DRSchlaubi) [reply gateway/src/main/kotlin/handler/HeartbeatHandler.kt:58]: I don't really know what the cause of this is though

Comment (DRSchlaubi) [reply gateway/src/main/kotlin/handler/HeartbeatHandler.kt:58]: I was able to find the cause of one of the two issues: Kotlin/kotlinx.atomicfu#180 (will now link it from the code)

Comment (DRSchlaubi) [reply gateway/src/main/kotlin/handler/HeartbeatHandler.kt]: Searching for the cause of the other bug I found actually two other bugs but could not track down which code introduces the bug i was looking for

**Comment by DRSchlaubi** [comment]
Looks like at least one of the issue will be fixed
https://github.com/Kotlin/kotlinx.atomicfu/issues/180#issuecomment-793843472

**Review by HopeBaron — APPROVED** [review]

**Pull request merged**